packages/vitest-plugin-storybook/src/global-setup.ts

```diff
@@ -0,0 +1,81 @@
+import { spawn, type ChildProcess } from 'node:child_process'
+import { log } from './utils'
+import http from 'node:http'
+import https from 'node:https'
+
+import type { GlobalSetupContext } from 'vitest/node'
+
+let storybookProcess: ChildProcess | null = null
+
+const checkStorybookRunning = (storybookUrl: string) => {
+  return new Promise<boolean>((resolve) => {
+    const url = new URL(`${storybookUrl}/iframe.html`)
+    const protocol = url.protocol === 'https:' ? https : http
+
+    const options = {
+      method: 'HEAD',
+      host: url.hostname,
+      path: url.pathname,
+    }
+
+    const req = protocol.request(options, (res) => {
+      resolve(res.statusCode === 200)
+    })
+
+    req.on('error', () => resolve(false))
+    req.end()
+  })
+}
+
+const startStorybookIfNeeded = async () => {
+  const storybookScript = process.env.__STORYBOOK_SCRIPT__ || ''
+  const storybookUrl = process.env.__STORYBOOK_URL__ || ''
+
+  const isRunning = await checkStorybookRunning(storybookUrl)
+
+  if (isRunning) {
+    log('Storybook is already running')
+    return
+  }
+
+  log(`Starting Storybook with command: ${storybookScript}`)
+
+  try {
+    // We do this sync because we don't want Vitest to hang while Storybook is starting
```

Comment:
What do you mean by this? Sync usually means it "hangs", while async processes don't because they use the event loop. It seems what you mean is that we don't await the spawn command, in order for Vitest to not wait for Storybook to start. If this were truly sync then I'd expect `spawnSync` to be used.

I wonder though, don't we need Vitest to wait for Storybook to start before it can run tests?

Reply:
Yea you got it! I just meant to spawn the process and not wait for it. Storybook is not needed at all for test to run, the idea is just to have it available once things fail, so users can click on a URL to access in Storybook